### PR TITLE
Commercial  f250 scrollable / parallax images.

### PR DESCRIPTION
**Pull request by austinh7**
@Calanthe For js checking!

This gives the ability for the responsive banner to have a parallax effect, (currently for top slot only).

scrolablef250

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/creatives/fluid250.js [hunk 1]
```diff
@@ -22,6 +24,24 @@ define([
...
+     * background-attachment - need to make this more granular
+     */
+    Fluid250.hasScrollEnabled = !detect.isIOS() && !detect.isAndroid();
+
+    Fluid250.prototype.updateBgPosition = function () {
+        var bgpositionx = this.params.backgroundPosition.split(' ')[0],
+        layeronebgpositionx = this.params.layerOneBGPosition.split(' ')[0],
+        layertwobgpositionx = this.params.layerTwoBGPosition.split(' ')[0],
+        layerthreebgpositionx = this.params.layerThreeBGPosition.split(' ')[0];
+
```
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/creatives/fluid250.js]: Minor thing: I would put all of those above variables in one vertical line.

File: static/src/javascripts/projects/common/modules/commercial/creatives/fluid250.js [hunk 2]
```diff
@@ -39,11 +59,21 @@ define([
                     '<iframe width="409px" height="230px" src="' + this.params.videoURL + '?rel=0&amp;controls=0&amp;showinfo=0&amp;title=0&amp;byline=0&amp;portrait=0" frameborder="0" class="fluid250_video fluid250_video--desktop fluid250_video--vert-pos-' + this.params.videoPositionV + ' fluid250_video--horiz-pos-' + this.params.videoPositionH + '" style="' + leftPosition + rightPosition + '"></iframe>' : ''
```
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/creatives/fluid250.js]: I think that this line is too long. Do you have any validation errors?

File: static/src/javascripts/projects/common/modules/commercial/creatives/fluid250.js [hunk 3]
```diff
@@ -39,11 +59,21 @@ define([
...
             };
 
-        $.create(template(fluid250Tpl, _.merge(this.params, templateOptions, videoDesktop))).appendTo(this.$adSlot);
+        this.$fluid250 = $.create(template(fluid250Tpl, _.merge(this.params, templateOptions, videoDesktop))).appendTo(this.$adSlot);
 
         if (this.params.trackingPixel) {
             this.$adSlot.before('<img src="' + this.params.trackingPixel + this.params.cacheBuster + '" class="creative__tracking-pixel" height="1px" width="1px"/>');
         }
+
+        if (Fluid250.hasScrollEnabled && (this.params.backgroundImageScroll !== 'normal' || this.params.layerOneImageScroll !== 'normal'  || this.params.layerTwoImageScroll !== 'normal'  || this.params.layerThreeImageScroll !== 'normal')) {
```
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/creatives/fluid250.js]: Too long line, I would suggest writing something like:
 if (Fluid250.hasScrollEnabled && (this.params.backgroundImageScroll !== 'normal' ||
    this.params.layerOneImageScroll !== 'normal' ||
    this.params.layerTwoImageScroll !== 'normal' ||
  (...)

File: static/src/javascripts/projects/common/modules/commercial/creatives/fluid250.js [hunk 4]
```diff
@@ -22,6 +24,34 @@ define([
...
+        if (this.params.layerTwoBGPosition !== '') {
+            layertwobgpositionx = this.params.layerTwoBGPosition.split(' ')[0];
+            $('.ad-slot--top-banner-ad .creative--fluid250 .fluid250_layer2').css('background-position', layertwobgpositionx + ' ' + window.pageYOffset * (this.params.layerTwoImageScroll / 100) + 'px').css('background-repeat', 'repeat-y');
+        }
+        if (this.params.layerThreeBGPosition !== '') {
+            layerthreebgpositionx = this.params.layerThreeBGPosition.split(' ')[0];
+            $('.ad-slot--top-banner-ad .creative--fluid250 .fluid250_layer3').css('background-position', layerthreebgpositionx + ' ' + window.pageYOffset * (this.params.layerThreeImageScroll / 100) + 'px').css('background-repeat', 'repeat-y');
+        }
+    };
+
```
Comment (phamann) [inline static/src/javascripts/projects/common/modules/commercial/creatives/fluid250.js]: Would be nice to see some FastDOM love here, i.e. I can see lots of reading an writing to the DOM.

**Comment by austinh7** [comment]
@phamann Can I get some help / review on the fastdom stuff? The code is all working but the tests are now failing.

File: static/src/javascripts/projects/common/modules/commercial/creatives/fluid250.js [hunk 5]
```diff
@@ -22,6 +26,45 @@ define([
...
 
+    /**
+     * TODO: rather blunt instrument this, due to the fact *most* mobile devices don't have a fixed
+     * background-attachment - need to make this more granular
+     */
+    Fluid250.hasScrollEnabled = !detect.isIOS() && !detect.isAndroid();
+
+    Fluid250.prototype.updateBgPosition = function () {
+
+        fastdom.read(function () {
```
Comment (phamann) [inline static/src/javascripts/projects/common/modules/commercial/creatives/fluid250.js]: Why do you have this wrapper? It has multiple `fastdom.writes` inside it?

File: static/test/javascripts/spec/common/commercial/creatives/fluid250.spec.js [hunk 6]
```diff
@@ -40,10 +42,16 @@ define([
...
-                videoURL: 'exampleVideoUrl'
+                videoURL: 'exampleVideoUrl',
+                backgroundPosition: 'left top',
+                layerOneBGPosition: 'left top',
+                layerTwoBGPosition: 'left top',
+                layerThreeBGPosition: 'left top'
             }).create();
-            expect(qwery('.fluid250_video--vert-pos-top', '.ad-slot').length).toBe(1);
-            expect(qwery('.fluid250_video--horiz-pos-left', '.ad-slot').length).toBe(1);
+            fastdom.defer(function () {
```
Comment (phamann) [inline static/test/javascripts/spec/common/commercial/creatives/fluid250.spec.js]: You will must likely need to pass in a number of frames to defer by: i.e.
```fastdom.defer(3, function() { ```

Comment (phamann) [inline static/test/javascripts/spec/common/commercial/creatives/fluid250.spec.js]: https://github.com/wilsonpage/fastdom#fastdomdeferframes-callback-context

**Comment by austinh7** [comment]
Closing temporarily until I get the expandable one sorted.

**Comment by austinh7** [comment]
This pull request either needs to go through or be taken on by someone before I leave.

Now tested with the sticky nav currently in A/B testing. It works but has some interesting results.

f250scrollstickynav

**Comment by phamann** [comment]
@Calanthe Do you know what the status of this is?

**Comment by Calanthe** [comment]
Patrick S. said that we should hold off reviewing this for now.

**Comment by kenlim** [comment]
OK, since this is up in the air right now, let's close this PR and work on the branch until it is ready.

**Pull request closed without merging**